src/base/static/components/place-detail/index.js

```diff
@@ -81,95 +72,87 @@ const PlaceDetailContainer = styled("div")(props => ({
 }));
 
 class PlaceDetail extends Component {
-  constructor(props) {
-    super(props);
-
-    this.surveyType = this.props.commentFormConfig.submission_type;
-    this.supportType = this.props.supportConfig.submission_type;
-    if (!this.props.model.submissionSets[this.surveyType]) {
-      this.props.model.submissionSets[
-        this.surveyType
-      ] = new SubmissionCollection(null, {
-        submissionType: this.surveyType,
-        placeModel: this.props.model,
-      });
-    }
-    if (!this.props.model.submissionSets[this.supportType]) {
-      this.props.model.submissionSets[
-        this.supportType
-      ] = new SubmissionCollection(null, {
-        submissionType: this.supportType,
-        placeModel: this.props.model,
-      });
-    }
-
-    this.categoryConfig = getCategoryConfig(
-      this.props.placeConfig,
-      this.props.model.get(constants.LOCATION_TYPE_PROPERTY_NAME),
-    );
-
-    this.state = {
-      placeModel: fromJS(this.props.model.attributes),
-      supportModels: serializeBackboneCollection(
-        this.props.model.submissionSets[this.supportType],
-      ),
-      surveyModels: serializeBackboneCollection(
-        this.props.model.submissionSets[this.surveyType],
-      ),
-      attachmentModels: serializeBackboneCollection(
-        this.props.model.attachmentCollection,
-      ),
-      isTagBarEditable: this.props.hasGroupAbilitiesInDatasets({
-        abilities: ["update", "destroy", "create"],
-        submissionSet: "tags",
-        datasetSlugs: [this.props.collectionId],
-      }),
-      isPlaceDetailEditable:
-        this.props.hasUserAbilitiesInPlace({
-          submitter: this.props.model.get(constants.SUBMITTER),
-          isSubmitterEditingSupported: this.categoryConfig
-            .submitter_editing_supported,
-        }) ||
-        this.props.hasGroupAbilitiesInDatasets({
-          abilities: ["update"],
-          submissionSet: "places",
-          datasetSlugs: [this.props.collectionId],
-        }),
-      isEditFormSubmitting: false,
-      isSurveyEditFormSubmitting: false,
-    };
-
-    // topOffset = header bar height + padding + geocoding bar height (if enabled).
-    this.topOffset = 80 + (this.props.mapConfig.geocoding_bar_enabled ? 72 : 0);
-  }
-
-  componentDidMount() {
-    // Maintain reset listeners for submission collections in case the detail
-    // view is instantiated before these collections have been fetched.
-    this.props.model.submissionSets[this.supportType].on(
-      "reset",
-      collection => {
-        this.setState({
-          supportModels: serializeBackboneCollection(collection),
-        });
-      },
-    );
-
-    this.props.model.submissionSets[this.surveyType].on("reset", collection => {
-      this.setState({
-        surveyModels: serializeBackboneCollection(collection),
-      });
+  state = {
+    place: fromJS(this.props.getPlace(this.props.placeId)),
```

Comment:
We are still storing a copy of the detail view's Place in local state, converted to immutable data structures.